### PR TITLE
feat!: UX improvements regarding agent activity

### PR DESCRIPTION
**Pull request by mohandast52**
### Please test all the different stages, including Start, Stop, agent running/pausing, etc.

- The last transaction made successfully

Scr2

- Tooltip info

Screen22M

- **NOTE**: 👇 This situation will occur if there's an error in fetching the transaction or if the blockNumber is too far behind, reaching a maximum of (latest - (10 * 9000)). In simpler terms, we can only go back 10 rounds, with each round covering 9000 blocks. 

Sc2

- Loading state

Scre1M

### Text changes

Sc1M

S222

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: frontend/components/MainPage/header/LastTransaction.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    height: 12px !important;
+    margin-top: 6px !important;
+  }
+`;
+
+const POLLING_INTERVAL = 60 * 1000; // 1 minute
+
+/**
+ * TODO:
+ * - add cache if the same block number is requested
```
Comment (Tanya-atatakai) [inline frontend/components/MainPage/header/LastTransaction.tsx:26]: maybe the result can be stored in storeState? and maybe not only the block number, but the last tx as well

File: frontend/components/MainPage/header/LastTransaction.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+export const LastTransaction = () => {
+  const { multisigAddress } = useAddress();
+
+  const [isFetching, setIsFetching] = useState(true);
+  const [transaction, setTransaction] = useState<TransactionInfo | null>(null);
+
+  const fetchTransaction = useCallback(async () => {
+    if (!multisigAddress) return;
+
+    getLatestTransaction(`${process.env.GNOSIS_RPC}`, multisigAddress)
```
Comment (Tanya-atatakai) [inline frontend/components/MainPage/header/LastTransaction.tsx:40]: does it make sense to pass the rpc into the getLatestTransaction? if we use gnosis anyway, can we just use `process.env.GNOSIS_RPC` inside the func?

File: frontend/service/Ethers.ts [hunk 3]
```diff
@@ -89,6 +90,65 @@ const checkRpc = async (rpc: string): Promise<boolean> => {
   }
 };
 
+const BACK_TRACK_BLOCKS = 9000;
```
[the remark this answers is not in the document]
Comment (Tanya-atatakai) [reply frontend/service/Ethers.ts:93]: wonder, why it works with no limitations on launch app where we request logs on gnosis using `usePublicClient` from `wagmi`, does that depend on the rpc?

**Review by Tanya-atatakai — APPROVED** [review]
looks really nice!

File: frontend/service/Ethers.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (truemiller) [reply frontend/service/Ethers.ts:93]: looks good to me!

just for more context/info on this:
**gnosis produces a block every 5 seconds, so 10000 block range is over 12 hours**

if feel like 10000 blocks is more than enough here, usually agents will transact for a few minutes to meet staking requirements

if the agent has not transacted in more than 12 hours, we can abstract this away by showing some text like `your agent has not transacted for while...`

File: frontend/components/MainPage/header/LastTransaction.tsx (hunk 1 above)
Comment (truemiller) [reply frontend/components/MainPage/header/LastTransaction.tsx:26]: storeState is only really for state that needs to be persisted i.e. has the user seen their first popup when the stake

gnosis produces a new block every 5 seconds, so should be perfectly fine to pull from chain every interval more than 5s

File: frontend/service/Ethers.ts [hunk 4]
```diff
@@ -89,6 +90,65 @@ const checkRpc = async (rpc: string): Promise<boolean> => {
...
+const MAX_ROUNDS = 10;
+
+/**
+ * Get the latest transaction details for the given contract address
+ */
+export const getLatestTransaction = async (
+  rpc: string,
+  contractAddress: Address,
+): Promise<TransactionInfo | null> => {
+  const provider = new providers.StaticJsonRpcProvider(rpc, {
```
Comment (truemiller) [inline frontend/service/Ethers.ts:103]: can reuse the provider from `constants/providers.ts` (probably belongs in `utils/...` if i'm honest)

File: frontend/service/Ethers.ts [hunk 5]
```diff
@@ -89,6 +90,65 @@ const checkRpc = async (rpc: string): Promise<boolean> => {
...
+    name: 'Gnosis',
+    chainId: 100, // we currently only support Gnosis Trader agent
+  });
+
+  const latestBlock = await provider.getBlockNumber();
+
+  const getLogsList = async (
```
Comment (truemiller) [inline frontend/service/Ethers.ts:110]: can we pull this into out of this function? may be useful elsewhere in function, and no need to recreate on every call

**Review by truemiller — APPROVED** [review]
two tiny changes, but not blocking

File: frontend/components/MainPage/header/LastTransaction.tsx (hunk 1 above)
Comment (Tanya-atatakai) [reply frontend/components/MainPage/header/LastTransaction.tsx:26]: okidok!
